src/topotoolbox/utils.py

```diff
@@ -0,0 +1,98 @@
+"""Provide general utility functions for topotoolbox.
+"""
+import sys
+import os
+
+from urllib.request import urlopen, urlretrieve
+
+from .grid_object import GridObject
+
+__all__ = ["load_dem", "get_dem_names", "read_tif"]
+
+DEM_SOURCE = "https://raw.githubusercontent.com/TopoToolbox/DEMs/master"
+DEM_NAMES = f"{DEM_SOURCE}/dem_names.txt"
+
+
+def read_tif(path: str) -> GridObject:
+    """Generate a new GridObject from a .tif file.
+
+    Args:
+        path (str): path to .tif file.
+
+    Returns:
+        GridObject: A new GridObject of the .tif file.
+    """
+    return GridObject(path)
```

Comment:
Would it make sense to read the data through rasterio in here and then use `read_tif` in the `GridObject` constructor? That might let us use the `GridObject` constructor to dispatch to different methods for loading data depending on the arguments provided to it. That might be a problem for another pull request though.

Reply:
Yes, I think that would be better. I think moving the `load_dem` to `grid_object.py` would be a good move. Also, moving `gen_random()`, [...] out of the class will improve usability. (calling `topotoolbox.gen_random()` is better than calling `topotoolbox.GridObject.gen_random())

Therefore, calling the `__init__()` of the GridObject should require passing all attributes and having them prepared beforehand.

These are quite a few changes, though, so that should definitely be done in a different pull request.

Also, just to add to that: while implementing this pull request, I was looking at a few other repositories like `pandas` and `seaborn`. Nobody seems to be using mixins. It seems to be convention to have very long files (2000 lines is not uncommon like [here](https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/categorical.py)). So after all, maybe we should get rid of them.
The only issue I see with that in our case is, that we are importing some files that only exist when building (`from ..grid import grid_identifyflats`). So my IDE is always yelling at me that some files don't exist. That's circumnavigated by using mixins.